### PR TITLE
Chore: Upgrade Go to 1.21.1

### PR DESCRIPTION
**Pull request by danielthedm**
Updates Grafana's Go version to 1.21.1.

### CODE REVIEW

**Comment by yuri-tceretian** [comment]
alerting related changes LGTM. Let's get a review from backend team

**Review by papagian — APPROVED** [review]
LGTM assuming that the changes are auto-generated by the script

**Comment by danielthedm** [comment]
No longer needed to address CVE.

**Pull request closed without merging**
